garrysmod/lua/includes/modules/widget.lua

```diff
@@ -32,13 +32,15 @@ local function UpdateHovered( pl, mv )
 	local trace =
 	{
 		start	= pl:EyePos(),
-		endpos	= pl:EyePos() + pl:GetAimVector() * 256,
+		endpos	= pl:GetAimVector(),
 		filter	= function( ent )
 
 			return IsValid( ent ) && ent:IsWidget()
 
 		end
 	}
+	trace.endpos:Mult( 256 )

```

Comment:
You need to do build your vector before the table. Doing it afterwards is horrible for readability.

Reply:
last edit: 7 Aug 2014. who reads this code?